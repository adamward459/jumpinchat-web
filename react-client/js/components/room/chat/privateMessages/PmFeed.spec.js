/* global jest, expect, describe, it, beforeEach */

import React from 'react';
import { shallow } from 'enzyme';
import PmFeed from './PmFeed.react';

describe('<PmFeed />', () => {
  let props;

  beforeEach(() => {
    props = {
      selectedConversation: null,
      chatInputValue: '',
      privateMessages: [],
      onSubmit: jest.fn(),
      onChange: jest.fn(),
      fixScroll: false,
    };
  });

  it('should not show the input if there is no selected conversation', () => {
    const wrapper = shallow(<PmFeed {...props} />);
    expect(wrapper.find('PmInput').length).toEqual(0);
  });

  it('should show the chat input if there is a selected conversation', () => {
    props.selectedConversation = 'foo';
    props.privateMessages = [
      {
        user: {
          userListId: 'foo',
          handle: 'bar',
        },
        messages: [],
      },
    ];
    const wrapper = shallow(<PmFeed {...props} />);
    expect(wrapper.find('PmInput').length).toEqual(1);
  });

  it('should render messages of selected conversation', () => {
    props.selectedConversation = 'foo';

    props.privateMessages = [
      {
        user: {
          userListId: 'foo',
          handle: 'bar',
        },
        messages: [
          {
            id: 'bar1',
            message: 'foo',

          },
          {
            id: 'bar2',
            message: 'foo',

          },
        ],
        disabled: false,
      },
      {
        user: { userListId: 'bar' },
        messages: [
          {
            id: 'baz1',
            message: 'foo',
          },
          {
            id: 'baz2',
            message: 'foo',
          },
        ],
        disabled: false,
      },
    ];

    const wrapper = shallow(<PmFeed {...props} />);
    expect(wrapper.getElement()).toMatchSnapshot();
  });
});
