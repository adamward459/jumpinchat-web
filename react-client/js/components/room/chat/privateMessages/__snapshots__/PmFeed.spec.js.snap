// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PmFeed /> should render messages of selected conversation 1`] = `
<div
  className="chat__FeedWrapper"
>
  <ForwardRef
    className="chat__Feed"
    contentStyle={
      Object {
        "paddingBottom": "0.25em",
      }
    }
    horizontal={false}
    onScroll={[Function]}
  >
    <RoomChatMessages
      currentUser={Object {}}
      fixScroll={false}
      messages={
        Array [
          Object {
            "id": "bar1",
            "message": "foo",
          },
          Object {
            "id": "bar2",
            "message": "foo",
          },
        ]
      }
      setScrollFixed={[Function]}
    />
  </ForwardRef>
  <PmInput
    disabled={false}
    onChange={[MockFunction]}
    onKeyDown={[Function]}
    onSendMessage={[MockFunction]}
    participant={
      Object {
        "handle": "bar",
        "userListId": "foo",
      }
    }
    value=""
  />
</div>
`;
